bin/compile

```diff
@@ -61,7 +61,7 @@ for BUILDPACK in $(cat $1/.buildpacks); do
       # echo "-----> Ls \$2: $(ls -la $2)"
       # echo "-----> Ls SUB_DIR(\$1/\$subdir): $(ls -la $1/$subdir)"
       # bin/compile BUILD_DIR CACHE_DIR ENV_DIR
-      $dir/bin/compile $1/$subdir $2 $3
+      $dir/bin/compile $1/$subdir $2/subdir/$subdir $3
```

Comment:
It is to prevent overlapping of cache dir.
We have `node` subdir, so buildpack for root dir and subdir will use same `$CACHE_DIR/node` directory.

- root: `$CACHE_DIR/node/cache/...
- node subdir: `$CACHE_DIR/node/node/cache/...`

While it looks safe for above case, cache directory structure is different for each buildpacks. So I added prefix for safety. :)

Reply:
@ypresto sorry for the late reply. That sounds good to me! I think as it stands now, the `subdir` is hard coded and not a variable (missing `$`). Was that intentional?

I think I prefer:
`node subdir: $CACHE_DIR/services/node/cache/...` rather than `/node/node` or `/subdir/node`

If you agree, we can change `subdir` --> `services` and then merge this in and release! (can you tag me @negativetwelve as well so I see this quicker, thanks!)